static/js/message_list_view.js

```diff
@@ -132,6 +132,59 @@ function update_message_date_divider(opts) {
         timerender.render_date(curr_time, prev_time, today)[0].outerHTML;
 }
 
+function clear_unread_messages_divider(msg) {
+    msg.want_unread_divider = false;
+    msg.unread_divider_html = undefined;
+}
+
+function add_unread_messages_divider(msg) {
+    msg.want_unread_divider = true;
+    msg.include_sender = true;
+
+    var NEW_MESSAGES_TEXT = 'NEW MESSAGES';
+
+    if (msg.want_date_divider) {
+        msg.want_date_divider = false;
+        var unread_divider_html = $(msg.date_divider_html);
+        unread_divider_html.find('.date-line').after(
+            NEW_MESSAGES_TEXT,
+            '<hr class="date-line">'
+        );
+        var timerender_class = unread_divider_html.attr('class');
+        msg.unread_divider_html = '<span class="' + timerender_class + '">' +
+                                  unread_divider_html.html() +
+                                  '</span>';
+    } else {
+        msg.unread_divider_html = '<span>' + NEW_MESSAGES_TEXT + '</span>';
+    }
+}
+
+function update_unread_messages_divider(opts) {
+    var prev_msg_container = opts.prev_msg_container;
+    var curr_msg_container = opts.curr_msg_container;
+    var filter = opts.list.data.filter;
+
+    if (!prev_msg_container) {
+        clear_unread_messages_divider(curr_msg_container);
+        return;
+    }
+
+    var is_topic = filter.has_operator('stream') && filter.has_operator('topic');
+    var is_pm = filter.has_operator('pm-with') || filter.has_operator('group-pm-with');
+
+    if (!(is_topic || is_pm)) {
```

Comment:
Let's add a comment before this check:
```
// Our unread message dividers, as currently conceived, don't really make sense in interleaved narrows,
// since we could have read messages after unread messages in those contexts.
// As a result, we currently don't offer these "New messages" notices in those views.
```

Reply:
(Just for context, I expect we'll later add support for this in a wider range of views, and e.g. just have the text change to something more subtle that communicates "New messages, including some messages you read via other views").